Sources/StreamChatUI/StreamSwiftyMarkdown/SwiftyMarkdown/SwiftyScanner.swift

```diff
@@ -259,20 +259,27 @@ class SwiftyScanner {
 		} else {
 			var remainingTags = min(openRange.upperBound - openRange.lowerBound, closeRange.upperBound - closeRange.lowerBound) + 1
 			while remainingTags > 0 {
-				if remainingTags >= self.rule.maxTags {
+				let shouldAppendStyle = remainingTags >= self.rule.maxTags
+				if shouldAppendStyle {
 					remainingTags -= self.rule.maxTags
 					if let style = self.rule.styles[ self.rule.maxTags ] {
 						if !styles.contains(where: { $0.isEqualTo(style)}) {
 							styles.append(style)
 						}
 					}
 				}
-				if let style = self.rule.styles[remainingTags] {
+
+				let remainingTagsStyle = self.rule.styles[remainingTags]
+				if let style = remainingTagsStyle {
 					remainingTags -= remainingTags
 					if !styles.contains(where: { $0.isEqualTo(style)}) {
 						styles.append(style)
 					}
 				}
+
+				if !shouldAppendStyle && remainingTagsStyle == nil {
+					break
+				}
```

Comment:
Can we also create a PR for the original repo? This sounds like a nice defensive additional to the original dependency

Reply:
Yep, will do that!

Reply:
https://github.com/SimonFairbairn/SwiftyMarkdown/pull/139